src/EnergyPlus/HeatRecovery.cc

```
@@ -364,107 +277,101 @@ namespace HeatRecovery {
                 }
             }
-            {
-                auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(3));
-                if (SELECT_CASE_var == "COUNTERFLOW") {
-                    state.dataHeatRecovery->ExchCond(ExchNum).FlowArr = HXConfiguration::CounterFlow;
-                } else if (SELECT_CASE_var == "PARALLELFLOW") {
-                    state.dataHeatRecovery->ExchCond(ExchNum).FlowArr = HXConfiguration::ParallelFlow;
-                } else if (SELECT_CASE_var == "CROSSFLOWBOTHUNMIXED") {
-                    state.dataHeatRecovery->ExchCond(ExchNum).FlowArr = HXConfiguration::CrossFlowBothUnmixed;
-                } else {
-                    ShowSevereError(state, cCurrentModuleObject + ": incorrect flow arrangement: " + state.dataIPShortCut->cAlphaArgs(3));
-                    ErrorsFound = true;
-                }
+
+            constexpr std::array<std::string_view, static_cast<int>(HXConfiguration::Num)> hxConfigurationNamesUC = {
+                "COUNTERFLOW", "PARALLELFLOW", "CROSSFLOWBOTHUNMIXED", "CROSS_FLOW_OTHER_NOT_USED"};
+            thisExchanger.FlowArr = static_cast<HXConfiguration>(getEnumerationValue(hxConfigurationNamesUC, state.dataIPShortCut->cAlphaArgs(3)));
+            if (thisExchanger.FlowArr == HXConfiguration::Invalid) {
+                ShowSevereError(state, cCurrentModuleObject + ": incorrect flow arrangement: " + state.dataIPShortCut->cAlphaArgs(3));
+                ErrorsFound = true;
             }
-            {
-                auto const SELECT_CASE_var(state.dataIPShortCut->cAlphaArgs(4));
-                if (SELECT_CASE_var == "YES") {
-                    state.dataHeatRecovery->ExchCond(ExchNum).EconoLockOut = EconomizerLockout::Yes;
-                } else if (SELECT_CASE_var == "NO") {
-                    state.dataHeatRecovery->ExchCond(ExchNum).EconoLockOut = EconomizerLockout::No;
+
+            BooleanSwitch toggle = getYesNoValue(state.dataIPShortCut->cAlphaArgs(4));
```

Aesthetically, I would do this idiom like:

```cpp
if (lAlphaFieldBlanks(4)) {
    thisExchanger.EconoLockOut = true;
else { 
    BooleanSwitch toggle = getYesNoValue(cAlphaArgs(4));
    if (toggle == BooleanSwitch::Invalid) {
       ShowSevereError;
    }
    thisEchanger.EconoLockOut = BooleanSwitch2Bool[static_cast<int>(toggle)];
}

---

Love it.

---

Got it.  Not sure where to drop BooleanSwitch2Bool array yet, but I've got this precedent started now at least.  Thanks.

---

Actually, I am dumb. You don't even need BooleanSwitch2Bool array, just `static_cast<bool>(toggle)`.

---

Fixed up these uses with this elegant pattern, thanks!

```
if (state.dataIPShortCut->lAlphaFieldBlanks(N)) {
	thisExchanger.EconoLockOut = true;
} else {
	BooleanSwitch toggle = getYesNoValue(state.dataIPShortCut->cAlphaArgs(N));
	if (toggle == BooleanSwitch::Invalid) {
		ShowSevereError(state, format("{}: incorrect econo lockout: {}", cCurrentModuleObject, state.dataIPShortCut->cAlphaArgs(N)));
	}
	thisExchanger.EconoLockOut = static_cast<bool>(toggle);
}
```

---

Why do you need the warning? Isn't that input a choice?

---

Hmm, I think it makes sense but I could be missing something:

- if the alpha field is blank then
  - we default to true
- else if the alpha field contains something then
  - we get an enum from that field, could be yes, no, or invalid (don't bring up input processor stuff please)
  - if the input is invalid then
    - we need to throw an error because of invalid input
  - otherwise just cast it to a true or false and be done

---

It's a choice field within min-fields so IP will fill with default if blank.

---

> - we get an enum from that field, could be yes, no, or invalid (**don't bring up input processor stuff please**)